### ISSUE TITLE
do not contain any metrics with the primary metric name at this moment, policy cannot be applied.

### ISSUE THREAD

**Issue by Carzyx** [Azure/MachineLearningNotebooks#1012]
Hi!
I'm working with Keras in Azure ML ([documentatnion](https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/training-with-deep-learning/train-hyperparameter-tune-deploy-with-keras/train-hyperparameter-tune-deploy-with-keras.ipynb)) and I'm getting a error when I try define a BanditPolicy to do a prevent train stop.

In the execution script I was define an accuracy primary_metric_name:

```
policy = BanditPolicy(evaluation_interval=2, slack_factor=0.000000001, delay_evaluation=10)
hdc = HyperDriveConfig(estimator=est,
                       hyperparameter_sampling=ps, 
                       policy=policy, 
                       primary_metric_name='Accuracy', 
                       primary_metric_goal=PrimaryMetricGoal.MAXIMIZE, 
                       max_total_runs=20,
                       max_concurrent_runs=1)

hdr = exp.submit(config=hdc)
```

And i was check that this param was be logged with a quick callback used in the train process

```
class CheckpointsCallback(Callback):
    def __init__(self, checkpoints_path):
        self.checkpoints_path = checkpoints_path

    def on_epoch_end(self, epoch, logs=None):
        # log a value repeated which creates a list
        run = Run.get_context()
        run.log('Loss', logs['val_loss'])
        run.log('Accuracy', logs['val_accuracy'])
        
        if self.checkpoints_path is not None:
            self.model.save_weights(self.checkpoints_path + "." + str(epoch))
            print("saved ", self.checkpoints_path + "." + str(epoch))
```

But.. I'm getting the next error:
**do not contain any metrics with the primary metric name at this moment, policy cannot be applied.**

Can anyone help me?
Thanks in advance

**Comment by SnehaAgrawal-MSFT** [comment]
@Carzyx Thanks for reaching here! Could you please share link of the document you are referring to help you better on this.

**Comment by RohitMungi-MSFT** [comment]
@Carzyx Could you please check if the value for accuracy needs to be logged as float. The example in the [documentation ](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-tune-hyperparameters#specify-an-early-termination-policy)for accuracy specifies this.

```
from azureml.core.run import Run
run_logger = Run.get_context()
run_logger.log("accuracy", float(val_accuracy))
```

**Comment by Carzyx** [comment]
> @Carzyx Thanks for reaching here! Could you please share link of the document you are referring to help you better on this.

I'm following this example.
https://github.com/Azure/MachineLearningNotebooks/blob/master/how-to-use-azureml/training-with-deep-learning/train-hyperparameter-tune-deploy-with-keras/train-hyperparameter-tune-deploy-with-keras.ipynb

**Comment by Carzyx** [comment]
> @Carzyx Could you please check if the value for accuracy needs to be logged as float. The example in the [documentation ](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-tune-hyperparameters#specify-an-early-termination-policy)for accuracy specifies this.
> 
> ```
> from azureml.core.run import Run
> run_logger = Run.get_context()
> run_logger.log("accuracy", float(val_accuracy))
> ```

It seems that number is a float actually.

**Comment by Carzyx** [comment]
For example, now I'm doing this process
```
from azureml.train.hyperdrive import RandomParameterSampling, BanditPolicy, HyperDriveConfig, PrimaryMetricGoal
from azureml.train.hyperdrive import choice, loguniform

ps = RandomParameterSampling(
    {
        '--optimizer_name': choice("SGD", "RMSprop", "Adam", "Adadelta", "Adagrad", "Adamax", "Nadam", "Ftrl")
    }
)
policy = BanditPolicy(evaluation_interval=1, **slack_factor=0.01,** delay_evaluation=5)
hdc = HyperDriveConfig(estimator=est,
                       hyperparameter_sampling=ps, 
                       policy=policy, 
                       primary_metric_name='Accuracy', 
                       primary_metric_goal=PrimaryMetricGoal.MAXIMIZE, 
                       max_total_runs=10,
                       max_concurrent_runs=1)

hdr = exp.submit(config=hdc)

from azureml.widgets import RunDetails
RunDetails(hdr).show()
```
In my screen I'm getting accuracy value correctly , but for some reason errors comes again and train doesn't stops when slack_factor is reached.

![image](https://user-images.githubusercontent.com/8115989/84757829-ecd42b80-afc4-11ea-8212-163505189dbb.png)

![image](https://user-images.githubusercontent.com/8115989/84757866-fcec0b00-afc4-11ea-8dde-f4aeb8ada692.png)

**Comment by swatig007** [comment]
@Carzyx can you please look at the log to see if it tried to cancel run hd_4a...3 (run id associated with 108) specially between intervals 6 and 10?

Based on the config, run 106 started forst and completed. Run 107 perfromed better than 106 around interval 6 after the delay_evaluation period and hence was not cancelled. Run 108 is the one where the policy should have applied, hence that's the one to check for,

@nagau FYI

**Comment by Carzyx** [comment]
Yes! It seems to work correctly.
I thought it was going wrong because the non-metric logs.
But I suppose it will be due to the 'job' that is listening if the metric has been launched.

Thanks all for your support!
